windows/deploy_vm/deploy_vm_from_iso.yml

```diff
@@ -4,19 +4,53 @@
 # Deploy a new Windows VM and install guest OS automatically
 # from ISO image
 #
+- name: "Set fact of the new VM default hardware version"
+  ansible.builtin.set_fact:
+    hardware_version: "{{ esxi_latest_hardware_version }}"
+  when: hardware_version is undefined
+
+- name: "Get '{{ guest_id }}' guest config options with VM hardware version '{{ hardware_version }}'"
+  include_tasks: ../../common/esxi_get_guest_config_options.yml
+  vars:
+    esxi_hardware_version: "{{ hardware_version }}"
+
 # Initialize VM virtual hardware config variables
 - name: "Initialize variables for new VM settings"
   ansible.builtin.set_fact:
     boot_disk_controller: >-
-      {%- if boot_disk_controller is defined and boot_disk_controller -%}{{ boot_disk_controller | lower }}
-      {%- elif esxi_cpu_vendor == 'arm' -%}nvme
-      {%- else -%}lsilogicsas
-      {%- endif -%}
+      {{
+         (boot_disk_controller is defined and boot_disk_controller)
+         | ternary(boot_disk_controller | lower, guest_config_options.default_disk_controller)
+      }}
+    network_adapter_type: >-
+      {{
+         (network_adapter_type is defined and network_adapter_type)
+         | ternary(network_adapter_type | lower, guest_config_options.default_ethernet)
+      }}
     secureboot_enabled: "{{ secureboot_enabled is defined and secureboot_enabled | lower == 'true' }}"
-    network_adapter_type: "{{ network_adapter_type | lower if (network_adapter_type is defined and network_adapter_type) else 'e1000e' }}"
-    firmware: "{{ firmware | lower if (firmware is defined and firmware) else 'efi' }}"
+    firmware: >-
+      {{
+         (firmware is defined and firmware)
+         | ternary(firmware | lower, guest_config_options.rec_firmware)
+      }}
     boot_disk_size_gb: "{{ [64, boot_disk_size_gb | default(64) | int] | max }}"
 
+- name: "Check VM boot disk controller is supported"
+  ansible.builtin.assert:
+    that:
+      - boot_disk_controller in guest_config_options.support_disk_controller
+    fail_msg: >-
+      New VM boot disk controller '{{ boot_disk_controller }}' is not in the disk controller
+      support list in guest config options '{{ guest_config_options.support_disk_controller }}'.
+
+- name: "Check new VM network adatper is supported"
+  ansible.builtin.assert:
+    that:
+      - network_adapter_type in guest_config_options.support_ethernet_card
+    fail_msg: >-
+      New VM network adatper '{{ network_adapter_type }}' is not in the Ethernet support list in guest
+      config options '{{ guest_config_options.support_ethernet_card }}'.
```

Comment:
How about moving these tasks for setting VM's hardware version, boot disk controller, ethernet, firmware, etc into a common task, which can be directly included in linux and windows? Or set these default values in env_setup for new_vm = true?